From 243778dfa631bf75aa11621f5b3dfb7c2ac6f2b7 Mon Sep 17 00:00:00 2001
From: Tom Hochstein <tom.hochstein@nxp.com>
Date: Tue, 21 Mar 2023 21:00:45 -0500
Subject: [PATCH] Revert EGL-X11 Plugin enablement change

Moving from 6.4 to 6.5.0 reveals a new build break:

```
tmp/work/armv8a-mx8-poky-linux/qtbase/6.5.0-r0/git/src/plugins/platforms/xcb/gl_integrations/xcb_egl/qxcbeglwindow.cpp:45:81: error: invalid conversion from 'xcb_window_t' {aka 'unsigned int'} to 'EGLNativeWindowType' {aka 'wl_egl_window*'} [-fpermissive]
   45 |     m_surface = eglCreateWindowSurface(m_glIntegration->eglDisplay(), m_config, m_window, nullptr);
      |                                                                                 ^~~~~~~~
      |                                                                                 |
      |                                                                                 xcb_window_t {aka unsigned int}
```

It seems that for 6.5.0, EGL-X11 Plugin is newly enabled. The enablement condition
was changed twice:

https://github.com/qt/qtbase/commit/c9df9f3976d969dc1f829a596016a377f1e619f4
https://github.com/qt/qtbase/commit/49d29440457a89bb0438ed882eb47f4f28fd328a

The original condition included qt_config_compile_test egl_x11, which has this
comment:

// Check if EGL is compatible with X. Some EGL implementations, typically on
// embedded devices, are not intended to be used together with X. EGL support
// has to be disabled in plugins like xcb in this case since the native display,
// window and pixmap types will be different than what an X-based platform
// plugin would expect.

Restore egl_x11 as the condition so that EGL-X11 Plugin is not enabled
for i.MX, as it was for Qt 6.4.

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Tom Hochstein <tom.hochstein@nxp.com>
---
 src/gui/configure.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gui/configure.cmake b/src/gui/configure.cmake
index d8319c120b..3527e1019f 100644
--- a/src/gui/configure.cmake
+++ b/src/gui/configure.cmake
@@ -932,7 +932,7 @@ qt_feature("xcb-glx" PRIVATE
 )
 qt_feature("xcb-egl-plugin" PRIVATE
     LABEL "EGL-X11 Plugin"
-    CONDITION QT_FEATURE_egl AND QT_FEATURE_opengl
+    CONDITION QT_FEATURE_egl_x11 AND QT_FEATURE_opengl
     EMIT_IF QT_FEATURE_xcb
 )
 qt_feature("xcb-native-painting" PRIVATE
-- 
2.25.1

